lib/application.ts

```diff
@@ -1107,29 +1110,40 @@ export class SNApplication {
     ephemeral = false,
     mergeLocal = true
   ) {
-    this.lockSyncing();
-    const result = await this.sessionManager.register(email, password);
-    if (!result.response.error) {
-      this.syncService!.resetSyncState();
-      await this.storageService!.setPersistencePolicy(
-        ephemeral
-          ? StoragePersistencePolicies.Ephemeral
-          : StoragePersistencePolicies.Default
-      );
-      if (mergeLocal) {
-        await this.syncService!.markAllItemsAsNeedingSync(true);
+    if (this.hasAccount()) {
+      throw Error('Tried to register when an account already exists.');
```

Comment:
Should we use `throw SNLog.error(...)` here instead?

Reply:
The way I see it `SNLog.error(...)` should be used for errors that are exceptional but do not warrant halting the current operation. Calling `application.signIn(...)` twice, on the other hand, is a logic fault that should be fixed by the client, hence the throw.
Any uncaught exception will be reported by our tools as well so this error wouldn't be lost in the ether if it happened on a user's device.